src/main/java/com/google/sps/dao/AssignmentDao.java

```diff
@@ -89,4 +94,112 @@ public ValidationResponse createAssignment(Assignment assignment) {
         }
         return assignments;
     }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForStudent(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
```

Comment:
id might not be present in newAssignment need to validate this before using newAssignment.getId(). Also if there is a validation error it should be returned before proceeding to get oldAssignment.

Reply:
done